core/src/main/java/com/amplifyframework/predictions/models/FaceLivenessSessionInformation.kt

```diff
@@ -20,6 +20,23 @@ import com.amplifyframework.annotations.InternalAmplifyApi
 data class FaceLivenessSessionInformation(
     val videoWidth: Float,
     val videoHeight: Float,
-    val challengeVersions: String,
-    val region: String
+    val challenge: String? = null,
+    val region: String,
+    val challengeVersions: List<Challenge>? = null,
+    val preCheckViewEnabled: Boolean? = null,
+    val attemptCount: Int? = null
```

Comment:
curious why `attemptCount` and `preCheckViewEnabled` are nullable here? is this for backward compatibility? does the UI component not pinned to an Amplify Android version?

Reply:
Its to make sure the constructor used by older versions of Amplify Liveness aren't broken if Amplify dependency is updated. While we resolve to a specific version, if a customer updates their Amplify dependency to a higher version, it will resolve to that one. 

We have a minimum Amplify version on the Liveness component.
Amplify Android must support current + all previous versions of Liveness

Reply:
Suggest something like this to make constructors cleaner at liveness call site.

```
@InternalAmplifyApi
class FaceLivenessSessionInformation {

    val videoWidth: Float
    val videoHeight: Float
    val challengeVersions: List<Challenge>
    val region: String
    val preCheckViewEnabled: Boolean?
    val attemptCount: Int?
    
    @Deprecated("Keeping compatibility for <= Amplify Liveness 1.2.6")
    constructor(
        videoWidth: Float,
        videoHeight: Float,
        challenge: String,
        region: String
    ) {
        this.videoWidth = videoWidth,
        this.videoHeight = videoHeight,
        this.challengeVersions = listOf(Challenge(ChallengeType.FaceMovementAndLightChallenge, challenge)),
        this.region = region
        this.preCheckViewEnabled = null
        this.attemptCount = null
    }

    constructor(
        videoWidth: Float,
        videoHeight: Float,
        region: String,
        challengeVersions: List<Challenge>,
        preCheckViewEnabled: Boolean,
        attemptCount: Int
    ) {
        this.videoWidth = videoWidth
        this.videoHeight = videoHeight
        this.region = region
        this.challengeVersions = challengeVersions
        this.preCheckViewEnabled = preCheckViewEnabled
        this.attemptCount = attemptCount
    }
}
```

If we can convert `challenge` the String to a list with a single item of `challengeVersions` it also makes downstream implementation easier.

Reply:
Ah this is a really good way to do this! Will update 

> If we can convert challenge the String to a list with a single item of challengeVersions it also makes downstream implementation easier.

I can make this change but wanted to better understand how this will make the downstream implementation easier?

Reply:
@vincetran Doing so would remove `challenge` entirely, so you no longer have to check if its null and attempt to use if present. If we do the conversion up front to `challengeVersions` then we have 1 field downstream to use.